Documentation/format/index.md

```diff
@@ -0,0 +1,205 @@
+│ └──────────────────┴──────────────────────────────────┘ │
+├─────────────────────────────────────────────────────────┤
+│ CRC32 <4b>                                              │
+└─────────────────────────────────────────────────────────┘
+```
+
+
+
+### Label Index
+
+The label index indexes holds lists of possible values for label names. Each label index can be a composite index over more than a single label name, which is tracked by `#names`, followed by the total number of entries.  
+The body holds `#entries` entries of possible values pointing back into the symbol table.
+
+```
+┌───────────────┬────────────────┬────────────────┐
+│ len <4b>      │ #names <4b>    │ #entries <4b>  │
+├───────────────┴────────────────┴────────────────┤
+│ ┌─────────────────────────────────────────────┐ │
+│ │ ref(value_0) <4b>                           │ │
+│ ├─────────────────────────────────────────────┤ │
+│ │ ...                                         │ │
+│ ├─────────────────────────────────────────────┤ │
+│ │ ref(value_n) <4b>                           │ │
+│ └─────────────────────────────────────────────┘ │
+│                      . . .                      │
+├─────────────────────────────────────────────────┤
+│ CRC32 <4b>                                      │
+└─────────────────────────────────────────────────┘
+```
+
+The sequence of label index sections is finalized by an offset table pointing to the beginning of each label index section for a given set of label names.
+
+### Postings
+
+Postings sections store monotinically increasing lists of series references that contain a given label pair associated with the list.  
+
+```
+┌────────────────────┬────────────────────┐
+│ len <4b>           │ #entries <4b>      │
+├────────────────────┴────────────────────┤
+│ ┌─────────────────────────────────────┐ │
+│ │ ref(series_1) <4b>                  │ │
+│ ├─────────────────────────────────────┤ │
+│ │ ...                                 │ │
+│ ├─────────────────────────────────────┤ │
+│ │ ref(series_n) <4b>                  │ │
+│ └─────────────────────────────────────┘ │
+├─────────────────────────────────────────┤
+│ CRC32 <4b>                              │
+└─────────────────────────────────────────┘
+```
+
+The sequence of postings sections is finalized by an offset table pointing to the beginning of each postings section for a given set of label names.
+
+### Offset Table
+
+An offset table stores a sequence of entries that maps a list of strings to an offset. They are used to track label index and postings sections. They are read into memory when an index file is loaded.
+
+```
+┌─────────────────────┬────────────────────┐
+│ len <4b>            │ #entries <4b>      │
+├─────────────────────┴────────────────────┤
+│ ┌──────────────────────────────────────┐ │
+│ │  n = #strs <uvarint>                 │ │
+│ ├──────────────────────┬───────────────┤ │
+│ │ len(str_1) <uvarint> │ str_1 <bytes> │ │
```

Comment:
Good question, in the end, there's no real answer I suppose. The tables are just very small so compression is really not needed. Sp complexity-wise, it seemed easier/more robust to just inline it if there's no benefit in using pointers.